sdk/core/azure-core/azure/core/paging.py

```diff
@@ -70,8 +72,12 @@ def __next__(self):
         # type: () -> Iterator[ReturnType]
         if self.continuation_token is None and self._did_a_call_already:
             raise StopIteration("End of paging")
+        try:
+            self._response = self._get_next(self.continuation_token)
+        except AzureError as error:
+            error.continuation_token = self.continuation_token
```

Comment:
Do we have to glue the `continuation_token` onto the exception here, or can we make sure we provide it when we raise the exception (e.g. by passing it into the context of the current operation)? It feels more fragile to catch/annotate/re-throw than to always pass in "what you know" about a request when making it...

Reply:
Off the top of my head I would say this would be tricky - as we don't have the continuation token context at the point of raising errors (at that point it's probably part of the URL, query parameters etc) - though it could probably be done by passing it in the pipeline context so I can investigate that.

Reply:
Yeah, no parsing of URLs. But if we can pass it in through the request context, custom pollers or pagers don't have to have the try/catch boiler-plate code. They *may* have to have the boiler-plate code to set the request context, though...

Reply:
@johanste  - on digging into this a little more, adding this to the error could be difficult because all the errors based on status code are raised in the generated code.
Unless we think it's unlikely that a status-code-based error would be continuable in the first place?

Reply:
Funnily enough @johanste, it seems to me more robust to do it that way, since it means we will set the continuation token whatever the way the callback is implemented (even badly). We're in the process in autorest to make this callable customizable, and I don't want to have people to have to think about that too much.

Once thing though @annatisch, if the continuation token is set already, because yes the callback did it, we shall not overwrite it.

About status code error, retry a 5xx is likely fair

Reply:
In addition to @lmazuel's comment - the exceptions are raised in the transport - so we would need to handle it separately in each transport (and consider it for any custom transport)

Reply:
As it is now, we are making a change to our protocols. So anybody who implements the protocol without deriving from our poller (which I don't know if any of our existing client libraries do) would presumably have to make the same. But if people are extending our base classes (which *is* - or at least should be - the recommended way), then we can have them covered.